SCons/Node/FS.py

```diff
@@ -92,6 +92,26 @@ def __str__(self):
                       repr(entry.name),
                       repr(self.attribute))
 
+
+class FileWhereDirectoryExpectedError(SCons.Errors.BuildError):
+    def __init__(self, node_path):
+        super().__init__()
+        self.node_path = node_path
+
+    def __str__(self):
+        fmt = "File %s found where directory expected."
+        return fmt % self.node_path
+
+
+class DirWhereFileExpectedError(SCons.Errors.BuildError):
```

Comment:
feels like a bit of overkill to define individual exceptions for this, but that's not a big deal.

Reply:
DirWhereFile and FileWhereDir result in different fixes/issues.. removing a file to put a dir is easy, but not vice versa.
Or at least that was the thought.